js/parser.js

```diff
@@ -50,33 +50,75 @@ var ERMrest = (function(module) {
+        var shortPath = (modifierPath === "" ? context.path : context.path.split(modifierPath)[0]); // path without modifiers
 
         // Parse out @sort(...) parameter and assign to context
         // Expected format:
         //  ".../catalog/catalog_id/entity/[schema_name:]table_name[/{attribute::op::value}{&attribute::op::value}*][@sort(column[::desc::])]"
-        if (uri.indexOf("@sort(") !== -1) {
+        if (modifierPath) {
+            if (modifierPath.indexOf("@sort(") !== -1) {
 
-            var sorts = path.match(/@sort\((.*)\)/)[1].split(",");
-            path = path.split("@sort(")[0];  // anything before @sort(..)
-            context.uri = uri.split("@sort(")[0]; // remove @sort from uri
+                var sorts = modifierPath.match(/@sort\(([^\)]*)\)/)[1].split(",");
 
-            context.sort = [];
-            for (var s = 0; s < sorts.length; s++) {
-                var sort = sorts[s];
-                var column = (sort.endsWith("::desc::") ?
-                    decodeURIComponent(sort.match(/(.*)::desc::/)[1]) : sort);
-                context.sort.push({"column": column, "descending": sort.endsWith("::desc::")});
+                context.sort = [];
+                for (var s = 0; s < sorts.length; s++) {
+                    var sort = sorts[s];
+                    var column = (sort.endsWith("::desc::") ?
+                        decodeURIComponent(sort.match(/(.*)::desc::/)[1]) : decodeURIComponent(sort));
+                    context.sort.push({"column": column, "descending": sort.endsWith("::desc::")});
+                }
+            }
+
+            // sort must specified to use @before and @after
+            if (modifierPath.indexOf("@before(") !== -1) {
+                if (context.sort) {
+                    context.paging = {};
+                    context.paging.before = true;
+                    context.paging.row = {};
+                    var row = modifierPath.match(/@before\(([^\)]*)\)/)[1].split(",");
+                    for (var i = 0; i < context.sort.length; i++) {
+                        // ::null:: to null, empty string to "", otherwise decode value
+                        var value = (row[i] === "::null::" ? null : (row[i] === "" ? "" : decodeURIComponent(row[i])));
+                        context.paging.row[context.sort[i].column] = value;
+                    }
+                }
+
+            } else if (modifierPath.indexOf("@after(") !== -1) {
+                if (context.sort) {
+                    context.paging = {};
+                    context.paging.before = false;
+                    context.paging.row = {};
+                    var row = modifierPath.match(/@after\(([^\)]*)\)/)[1].split(",");
+                    for (var i = 0; i < context.sort.length; i++) {
+                        // ::null:: to null, empty string to "", otherwise decode value
+                        var value = (row[i] === "::null::" ? null : (row[i] === "" ? "" : decodeURIComponent(row[i])));
```

Comment:
I see I commented on an outdated diff, but the same line is still here... Can't `(row[i] === "" ? "" : decodeURIComponent(row[i]))` be simply `decodeURIComponent(row[i])` with exactly the same effect, both here and in the previous conditional block?